src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -174,6 +174,44 @@ public void RegisterOperationAction(Action<OperationAnalysisContext> action, par
         /// Hence, end actions are never executed concurrently with non-end actions operating on the same analysis unit.
         /// </remarks>
         public abstract void EnableConcurrentExecution();
+
+        /// <summary>
+        /// Configure analysis mode of generated code for this analyzer.
+        /// Non-configured analyzers will default to <see cref="GeneratedCodeAnalysisFlags.Default"/> mode for generated code.
+        /// It is recommended for the analyzer to always invoke this API with the required <see cref="GeneratedCodeAnalysisFlags"/> setting.
+        /// </summary>
+        public abstract void ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags analysisMode);
+    }
+
+    /// <summary>
+    /// Flags to configure mode of generated code analysis.
+    /// </summary>
+    [Flags]
+    public enum GeneratedCodeAnalysisFlags
+    {
+        /// <summary>
+        /// Disable analysis and diagnostic reporting for generated code.
+        /// </summary>
+        None = 0x00,
+
+        /// <summary>
+        /// Enable analysis for generated code.
+        /// </summary>
+        Analyze = 0x01,
```

Comment:
Updated the doc comments. I had initially thought of calling this field "AnalyzerActionCallbacks" but that seemed too verbose.